src/data/matrix/block.lean

```diff
@@ -213,16 +213,51 @@ by { ext i, cases i; simp [vec_mul, dot_product] }
 
 variables [decidable_eq l] [decidable_eq m]
 
-@[simp] lemma from_blocks_diagonal [has_zero α] (d₁ : l → α) (d₂ : m → α) :
+section has_zero
+variables [has_zero α]
+
+lemma to_block_diagonal_self (d : m → α) (p : m → Prop) :
+  matrix.to_block (diagonal d) p p = diagonal (λ i : subtype p, d ↑i) :=
+begin
+  ext i j,
+  by_cases i = j,
+  { simp [h] },
+  { simp [has_one.one, h, λ h', h $ subtype.ext h'], }
+end
+
+lemma to_block_diagonal_disjoint (d : m → α) {p q : m → Prop} (hpq : disjoint p q) :
+  matrix.to_block (diagonal d) p q = 0 :=
+begin
+  ext ⟨i, hi⟩ ⟨j, hj⟩,
+  have := hpq i,
```

Comment:
This looks unused?

Reply:
oops, good that you are paying attention :-)